video/out/gpu/context.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include "video/out/vo.h"
+
+#include "config.h"
+#include "ra.h"
+
+enum ra_api_type {
+    RA_API_ANY    = 0,
+    RA_API_OPENGL,
+};
+
+struct ra_ctx_opts {
+    int allow_sw;      // allow software renderers
+    int want_alpha;    // create an alpha framebuffer if possible
+    int debug;         // enable debugging layers/callbacks etc.
+    bool probing;      // the backend was auto-probed
+    int swchain_depth; // max number of images to render ahead
```

Comment:
What if you want to change these at runtime?

Reply:
That would be possible in theory but I didn't want to have to worry about it. So for now it's assumed to be constant. It wouldn't be much of an incremental step to add a `resize_swapchain` function which we can use whenever this option changes, but it would require special care from the backends that use external swapchains and I'm not really capable of touching that code.

The vulkan and opengl code can already handle a dynamic swapchain size. The context_vdpau.c code would need to be made slightly smarter. I consider it out-of-scope of this PR, which is already too big as it is.